src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Kafka/ITimestamp.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Kafka
+{
+    /// <summary>
+    /// Timestamp struct for duck-typing
+    /// Requires boxing, but necessary as we need to duck-type <see cref="Type"/> too
+    /// </summary>
+    public interface ITimestamp
+    {
+        /// <summary>
+        /// Gets the timestamp type
+        /// </summary>
+        public TimestampType Type { get; }
```

Comment:
If we declared this to return an `int`, would that allow us to do any additional optimizations?

Reply:
I thought I'd tried that but I couldn't get the duck typing to work 🤔  I'll give it another try

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1444/commits/b1419d91627e7511ed11be120367c23809606ff1